skynet/modules/ttt/rag/stores/faiss.py

```diff
@@ -62,14 +63,20 @@ async def create(self, store_id, documents):
         index = faiss.IndexFlatL2(len(self.embedding.embed_query(store_id)))
         vector_store = FAISS(
             embedding_function=self.embedding,
-            distance_strategy=DistanceStrategy.EUCLIDEAN_DISTANCE,
+            distance_strategy=DistanceStrategy.COSINE,
             index=index,
             docstore=InMemoryDocstore(),
             index_to_docstore_id={},
+            normalize_L2=True,
```

Comment:
What does this do?

Reply:
cosine is the recommended strategy for text search, however, we need to eliminate from its calculation the effect that the length of the vectors have on the result, because for similarity we only need the direction. This is where normalisation comes in